reedsolomon.go

```diff
@@ -65,7 +65,7 @@ type Encoder interface {
 	// Given a list of shards, some of which contain data, fills in the
 	// data shards that don't have data.
 	//
-	// The length of the array must be equal to Shards.
+	// The length of the array must be equal to TotalShards.
```

Comment:
You cannot change the interface.

You must accept *either* Shards or TotalShards and send it to the appropriate function.

Reply:
It was unintended change to unrelated ReconstructData doc. Reverted it.

Reply:
@walldiss I mean you are changing the behavior of the interface.  This will break existing users.

It is fine if you change the behavior based on the length of the provided slice, so the user can chose behavior based on that. So changes must be transparent to existing users and you must provide documentation on the difference.

Reply:
Got it. I've modified code to behave exactly same as it was with old usage semantics. 

One thing that worries me a bit is that implementations of `ReconstructSome` did not have any validation, that enforced size of `required` array to be equal to datashards, as required by interface. Which could mean, that there is a possibility, that there are some users that could have used it incorrectly by passing array with size of all shards and thus will be affected by this change.